Library/Services/Implementation/NfieldCapiInterviewersService.cs

```diff
@@ -0,0 +1,223 @@
+//    the Free Software Foundation, either version 3 of the License, or
+//    (at your option) any later version.
+//
+//    Nfield.SDK is distributed in the hope that it will be useful,
+//    but WITHOUT ANY WARRANTY; without even the implied warranty of
+//    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//    GNU Lesser General Public License for more details.
+//
+//    You should have received a copy of the GNU Lesser General Public License
+//    along with Nfield.SDK.  If not, see <http://www.gnu.org/licenses/>.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using Nfield.Extensions;
+using Nfield.Infrastructure;
+using Nfield.Models;
+using Nfield.Models.NipoSoftware.Nfield.Manager.Api.Models;
+using Nfield.Quota.Helpers;
+
+namespace Nfield.Services.Implementation
+{
+    /// <summary>
+    /// Implementation of <see cref="INfieldCapiInterviewersService"/>
+    /// </summary>
+
+    internal class NfieldCapiInterviewersService : INfieldCapiInterviewersService, INfieldConnectionClientObject
+    {
+        #region Implementation of INfieldCapiInterviewersService
+
+        /// <summary>
+        /// See <see cref="INfieldCapiInterviewersService.AddAsync"/>
+        /// </summary>
+        public Task<InterviewerChanged> AddAsync(Interviewer interviewer)
+        {
+            return Client.PostAsJsonAsync(CapiInterviewersApi, interviewer)
+                         .ContinueWith(task => task.Result.Content.ReadAsStringAsync().Result)
+                         .ContinueWith(task => JsonConvert.DeserializeObject<InterviewerChanged>(task.Result))
+                         .FlattenExceptions();
+        }
+
+        /// <summary>
+        /// See <see cref="INfieldCapiInterviewersService.RemoveAsync"/>
+        /// </summary>
+        public Task RemoveAsync(Interviewer interviewer)
+        {
+            if (interviewer == null)
+            {
+                throw new ArgumentNullException("interviewer");
+            }
+
+            return
+                Client.DeleteAsync(new Uri(CapiInterviewersApi, interviewer.InterviewerId))
+                      .FlattenExceptions();
+        }
+
+        /// <summary>
+        /// See <see cref="INfieldCapiInterviewersService.UpdateAsync"/>
+        /// </summary>
+        public Task<InterviewerChanged> UpdateAsync(Interviewer interviewer)
+        {
+            if (interviewer == null)
+            {
+                throw new ArgumentNullException("interviewer");
+            }
+
+            var updatedInterviewer = new UpdateCapiInterviewer
+            {
+                EmailAddress = interviewer.EmailAddress,
+                FirstName = interviewer.FirstName,
+                LastName = interviewer.LastName,
+                TelephoneNumber = interviewer.TelephoneNumber,
+                IsSupervisor = interviewer.IsSupervisor
+            };
+
```

Comment:
You can use an anonymous class and remove the internal class UpdateCapiInterviewer
```suggestion
            var updatedInterviewer = new
            {
                EmailAddress = interviewer.EmailAddress,
                FirstName = interviewer.FirstName,
                LastName = interviewer.LastName,
                TelephoneNumber = interviewer.TelephoneNumber,
                IsSupervisor = interviewer.IsSupervisor
            };

```

Reply:
But it's better for the testing, so you expect a certain class in the mock class